app/assets/javascripts/preferences/panes/backups-segments/EmailBackups.tsx

```diff
@@ -0,0 +1,204 @@
+  const [emailFrequency, setEmailFrequency] = useState<EmailBackupFrequency>(
+    EmailBackupFrequency.Weekly
+  );
+  const [emailFrequencyOptions, setEmailFrequencyOptions] = useState<
+    DropdownItem[]
+  >([]);
+  const [isFailedBackupEmailMuted, setIsFailedBackupEmailMuted] =
+    useState(false);
+  const [isFailedCloudBackupEmailMuted, setIsFailedCloudBackupEmailMuted] =
+    useState(false);
+  const [isEntitledForEmailBackups, setIsEntitledForEmailBackups] =
+    useState(false);
+
+  const convertBooleanStringToBoolean = (value: string) => {
+    return value !== 'false';
+  };
+
+  useEffect(() => {
+    const emailBackupsFeatureStatus = application.getFeatureStatus(
+      FeatureIdentifier.DailyEmailBackup
+    );
+    setIsEntitledForEmailBackups(
+      emailBackupsFeatureStatus === FeatureStatus.Entitled
+    );
+
+    const loadEmailFrequencySetting = async () => {
+      setIsLoading(true);
+
+      try {
+        const userSettings = await application.listSettings();
+        setEmailFrequency(userSettings.EMAIL_BACKUP as EmailBackupFrequency);
+        setIsFailedBackupEmailMuted(
+          convertBooleanStringToBoolean(
+            userSettings[SettingName.MuteFailedBackupsEmails] as string
+          )
+        );
+        setIsFailedCloudBackupEmailMuted(
+          convertBooleanStringToBoolean(
+            userSettings[SettingName.MuteFailedCloudBackupsEmails] as string
+          )
+        );
+      } catch (error) {
+        console.log(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    const frequencyOptions = [];
+    for (const frequency in EmailBackupFrequency) {
+      const frequencyValue =
+        EmailBackupFrequency[frequency as keyof typeof EmailBackupFrequency];
+      frequencyOptions.push({
+        value: frequencyValue,
+        label: capitalizeFirstLetter(frequencyValue),
+      });
+    }
+    setEmailFrequencyOptions(frequencyOptions);
+
+    loadEmailFrequencySetting();
+  }, [application]);
+
+  const updateUserSetting = async (
+    settingName: SettingName,
+    payload: string
+  ) => {
+    try {
+      await application.updateSetting(settingName, payload);
+    } catch (e) {
+      application.alertService.alert(STRING_FAILED_TO_UPDATE_USER_SETTING);
+    }
+  };
+
+  const updateEmailFrequency = async (frequency: EmailBackupFrequency) => {
+    setEmailFrequency(frequency);
+
+    await updateUserSetting(SettingName.EmailBackup, frequency);
+  };
+
+  const toggleMuteFailedBackupEmails = async () => {
+    setIsFailedBackupEmailMuted(!isFailedBackupEmailMuted);
+    await updateUserSetting(
+      SettingName.MuteFailedBackupsEmails,
+      `${!isFailedBackupEmailMuted}`
+    );
+  };
+
+  const toggleMuteFailedCloudBackupEmails = async () => {
+    setIsFailedCloudBackupEmailMuted(!isFailedCloudBackupEmailMuted);
+
+    await updateUserSetting(
+      SettingName.MuteFailedCloudBackupsEmails,
+      `${!isFailedCloudBackupEmailMuted}`
+    );
+  };
```

Comment:
Would it be helpful to move the business logic to a MobX store?

It looks like the component deals with quite a lot of business-related logic (not just `useState` to hold the content of text input but also a lot of state related to async operation with the data, loading states, etc),

Usually that's the sign that a mobx store could help separate concerns (business vs interface) and make your react component simpler.

A few refs:

https://mobx.js.org/react-integration.html
https://developer.mozilla.org/en-US/docs/Glossary/MVC